lib/bundler/rubygems_gem_installer.rb

```diff
@@ -31,12 +31,24 @@ def validate_bundler_checksum(checksum)
         send(checksum_type(checksum), digest)
       end
       unless digest == checksum
-        raise SecurityError,
-          "The checksum for the downloaded `#{spec.full_name}.gem` did not match " \
-          "the checksum given by the API. This means that the contents of the " \
-          "gem appear to be different from what was uploaded, and could be an indicator of a security issue.\n" \
-          "(The expected SHA256 checksum was #{checksum.inspect}, but the checksum for the downloaded gem was #{digest.inspect}.)\n" \
-          "Bundler cannot continue installing #{spec.name} (#{spec.version})."
+        raise SecurityError, <<-MESSAGE
```

Comment:
Doing something like that breaks the formatting because it either removes the empty newlines required for spacing out the message for readability, or it doesn't remove the leading spaces in the paragraphs. Heredoc actually seems to handle removing the leading spaces correctly but I can't be certain of all ruby versions and platforms... the test will fail if there are leading spaces in the paragraphs.